sktime/datasets/_single_problem_loaders.py

```diff
@@ -1740,3 +1742,102 @@ def create_series_data(df, cal, sp, include_events=False, test=False):
             inplace=True,
         )
         return sales_train_validation, sell_prices, calendar
+
+
+def load_stallion(special_dates=True, verbose=False):
+    """Load the Stallion Volume Forecasting for forecasting with exogenous variables.
+
+    It's a simplified version of the original [Stallion kaggle competition](https://www.kaggle.com/datasets/utathya/future-volume-prediction).
+
+
+    Parameters
+    ----------
+    special_dates : bool, default=True
+        include special datas columns
+    verbose : bool, default=False
+        print downloading status to console
+
+    Returns
+    -------
+    y: pd.DataFrame
+        The target series to be predicted.
+    X: pd.DataFrame
+        The exogenous time series data for the problem.
+
+    Examples
+    --------
+    >>> from sklearn.model_selection import train_test_split
+
+    >>> from sktime.datasets import load_stallion
+
+    >>> y, X = load_stallion()
+    >>> X_train, X_test, y_train, y_test = train_test_split(
+            X, y, test_size=0.1, train_size=0.9, shuffle=False
+        )
+    """
+    name = "Stallion"
+    fname = name + ".csv"
+    data_dir = os.path.join(MODULE, "local_data", name)
+    _mkdir_if_not_exist(data_dir)
+    path = os.path.join(data_dir, fname)
+    if not os.path.exists(path):
+        from urllib.request import urlretrieve
+
+        parquet_file = os.path.join(data_dir, name + ".parquet")
+        u = "https://github.com/jdb78/pytorch-forecasting/raw/master/examples/data/stallion.parquet"
+        repeats = 5
+        for repeat in range(repeats):
+            try:
+                urlretrieve(
+                    url="https://github.com/jdb78/pytorch-forecasting/raw/master/examples/data/stallion.parquet",
+                    filename=parquet_file,
+                )
+                break
+            except Exception as e:
+                if verbose:
+                    print(  # noqa: T201
+                        f"Downloading dataset {name} from {u} to {parquet_file}"
+                        f"(attempt {repeat+1} of {repeats} total failed). "
+                    )
+                if repeat + 1 == repeats:
+                    raise e
+
+        data = pd.read_parquet(parquet_file)
```

Comment:
I've a small question for this line: if I'm not wrong, this needs pyarrow. Is it part of sktime's core dependencies, or implied by them?

Reply:
`pyarrow` is not included or implied by core dependencies. Should I add it to core dependencies or `datasets` dependencies maybe?

Reply:
I don't think we should modify core dependencies, but dataset one you can do. Also, in the first place, I think you should add some checks in this module for presence of pyarrow.

If I'm not wrong, @pranavvp16 is also directly or indirectly using pyarrow for polars. And pandas will use pyarrow as core dependency starting 3.0. So, given these, @fkiraly / @benHeid do you already have any plans for pyarrow support in sktime?